### PR TITLE
[FIX] website_slides_survey: stop showing not finished attempts

### PR DESCRIPTION
**Pull request by Ephoniz**
## Issue:

Right now we will be showing as Attemtps (under the website profile in attempts tab) any attempt, even whe we didn't even started the certification yet.

## Steps to reproduce:

1. Install website_slide_survey.
2. Create or go to an existing course logged in.
3. Create or use an exsiting certigication and press the 'Begin certification' and don't start the certification after the redirect.
4. Go to your profile and check for the attemtps.

## Solution:

It will make more sense that we only show the actual attempts, which means that we shouldn't show here the attempts that are not finished, since how user_inputs works and are always created at the time of the link creation, we could just filter out the user_inputs that are not actually valid to show here and show only the proper ones.

opw-3781323

### CODE REVIEW

File: addons/website_slides_survey/views/website_profile.xml [hunk 1]
```diff
@@ -13,7 +13,8 @@
         <xpath expr="//ul[hasclass('o_wprofile_nav_tabs')]" position="inside">
             <li t-if="show_certification_tab" class="nav-item">
                 <a role="tab" aria-controls="certification" href="#profile_tab_content_certification" t-attf-class="nav-link #{ 'active' if active_tab == 'certification' else '' }" data-bs-toggle="tab">
-                    <t t-if="user_inputs" t-out="len(user_inputs)" /> Certification Attempts
+                    <t t-if="user_inputs" t-set="certification_attempts" t-value="len([user_input for user_input in user_inputs if user_input['state'] == 'done'])"/>
```
Comment (nle-odoo) [inline addons/website_slides_survey/views/website_profile.xml:16]: we could remove [ and ] to have a generator

```suggestion
                    <t t-if="user_inputs" t-set="certification_attempts" t-value="len(user_input for user_input in user_inputs if user_input['state'] == 'done')"/>
```
(same things below)

**Review by nle-odoo — APPROVED** [review]
looks good to me

there is some risk with changing the XML structure like this (the t-out/t-foreach/t-if attribute) in case someone used it for a XPATH, but I don't see much way around it

technically it's a little uglier but we could do:

```py
<t t-set="user_inputs" t-value="user_input.filtered(lambda i: i['state'] == 'done')"/>
```
and leave the code as it is currently, but if it was important for other code to have the full list, we would need to keep the original value then set it back after our usage

note: if in all case we wanted them filtered, maybe it mike make sense to already give the list filtered to the view

File: addons/website_slides_survey/views/website_profile.xml (hunk 1 above)
Comment (flch-odoo) [reply addons/website_slides_survey/views/website_profile.xml]: > there is some risk with changing the XML structure like this

Indeed. How about upating `WebsiteSlidesSurvey._prepare_user_profile_values` to add `('state', '=', 'done')` in the domain instead? I don't think we ever need to show the other attempts.

> note: if in all case we wanted them filtered, maybe it mike make sense to already give the list filtered to the view

= this ^^

**Review by flch-odoo — CHANGES_REQUESTED** [review]
https://github.com/odoo/odoo/pull/161456#discussion_r1624620427

**Comment by Ephoniz** [comment]
@flch-odoo thank you for the review, you're right this solution is way simpler and safer (sorry for the delay on fixing this 🙂) It should be good now.

**Review by flch-odoo — APPROVED** [review]
Thank you!

**Comment by kitan191** [comment]
@robodoo delegate+

**Comment by Ephoniz** [comment]
robodoo r+

**Pull request closed without merging**